src/main/python/mind_your_ps_and_qs.py

```diff
@@ -1,16 +1,20 @@
 import requests
 
 
-def get_ps_and_qs(ps_and_qs, factors, idx, curr_factor_one, curr_factor_two):
-    if idx == len(factors):
-        if curr_factor_one == 1 or curr_factor_two == 1:
-            return
+def get_ps_and_qs(coprimes):
+    ps_and_qs = []
+    get_ps_and_qs_helper(ps_and_qs, coprimes, 0, 1, 1)
+    return ps_and_qs
+
+
+def get_ps_and_qs_helper(ps_and_qs, coprimes, idx, curr_factor_one, curr_factor_two):
+    if idx == len(coprimes):
         ps_and_qs.append({curr_factor_one, curr_factor_two})
         return
-    factor_one_new = curr_factor_one * pow(int(factors[idx][0]), int(factors[idx][1]))
-    get_ps_and_qs(ps_and_qs, factors, idx + 1, factor_one_new, curr_factor_two)
-    factor_two_new = curr_factor_two * pow(int(factors[idx][0]), int(factors[idx][1]))
-    get_ps_and_qs(ps_and_qs, factors, idx + 1, curr_factor_one, factor_two_new)
+    factor_one_new = curr_factor_one * pow(int(coprimes[idx][0]), int(coprimes[idx][1]))
```

Comment:
In the recursive helper method, you don't need to save both `ps` and `qs`. Just keep on list for either `p` or `q`. you can get the other side in the `get_ps_and_qs` method by doing division. It will looks much cleaner.

Reply:
agree